automated/linux/imagenet-resnet50/imagenet-resnet50.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+# shellcheck disable=SC1091
+
+. ../../lib/sh-test-lib
+OUTPUT="$(pwd)/output"
+
+TEST_GIT_URL="https://github.com/mlcommons/inference.git"
+TEST_DIR="$(pwd)/${TEST_PROGRAM}"
+TEST_PROG_VERSION="215c057fc6690a47f3f66c72c076a8f73d66cb12"
+TEST_PROGRAM="inference"
+
+usage() {
+    echo "Usage: $0 [-a <home-directory>]
+                    [-m <mount-directory>]
+                    [-t <true|false>]
+                    [-p <test-dir>]
+                    [-v <test-prog-version>]
+                    [-u <test-git-url>]
+                    [-s <true|false>]" 1>&2
+    exit 1
+}
+
+while getopts "a:m:t:s:p:v:u:" o; do
+    case "$o" in
+        a) HOME_DIR="${OPTARG}";;
+        m) MNT_DIR="${OPTARG}" ;;
+        t) MNT_EXISTS="${OPTARG}" ;;
+        s) SKIP_INSTALL="${OPTARG}" ;;
+        p) TEST_DIR="${OPTARG}" ;;
+        v) TEST_PROG_VERSION="${OPTARG}" ;;
+        u) TEST_GIT_URL="${OPTARG}" ;;
+        *) usage ;;
+    esac
+done
+
+pkgs="build-essential git python3-venv python3-dev libhdf5-dev gcc-aarch64-linux-gnu curl"
+install_deps "${pkgs}" "${SKIP_INSTALL}"
+create_out_dir "${OUTPUT}"
+rm -rf "${HOME_DIR}"/tf_venv
+rm -rf "${HOME_DIR}"/src
+rm -rf "${MNT_DIR}"/datasets
+mkdir "${HOME_DIR}"/tf_venv
+pushd "${HOME_DIR}"/tf_venv || exit
+python3 -m venv .
+source bin/activate
+popd || exit
+
+if [ "${SKIP_INSTALL}" = false ]; then
+    pushd "${HOME_DIR}"/tf_venv || exit
+    python -m pip install --upgrade pip wheel
+    python -m pip install h5py
+    python -m pip install cython
+    python -m pip install google protobuf==3.20.1
+    python -m pip install --no-binary pycocotools pycocotools
+    python -m pip install absl-py pillow
+    python -m pip install --extra-index-url https://snapshots.linaro.org/ldcg/python-cache/ numpy==1.19.5
+    python -m pip install --extra-index-url https://snapshots.linaro.org/ldcg/python-cache/ matplotlib
+    python -m pip install ck
+    ck pull repo:ck-env
+    python -m pip install scikit-build
+    python -m pip install --extra-index-url https://snapshots.linaro.org/ldcg/python-cache/ tensorflow-io-gcs-filesystem==0.21.0 h5py==3.1.0
```

Comment:
isn't most of these steps in the skip_install identical in all the different script files?
would it make sense to have a function for this?

Reply:
You're correct, I think most of the setup steps in the skip_install are identical and the only differences are when getting the dataset in MNT_EXISTS. Would you prefer I created a function for this or is it alright to leave as is?